internal/services/applicationinsights/application_insights_resource.go

```diff
@@ -253,7 +244,7 @@ func resourceApplicationInsightsCreateUpdate(d *pluginsdk.ResourceData, meta int
 	if err != nil {
 		return fmt.Errorf("retrieving %s: %+v", id, err)
 	}
-	if read.Model == nil && read.Model.Id == nil {
+	if read.Model == nil || read.Model.Id == nil {
 		return fmt.Errorf("cannot read %s", id)
 	}
```

Comment:
I think this could benefit from some more specific/split error messages

```suggestion
	if read.Model == nil {
		return fmt.Errorf("retrieving %s: `model` was nil", id)
	}
	
	if read.Model.Id == nil {
		return fmt.Errorf("retrieving %s: `id` was nil", id)
	}
```